Move Repeat<t> to an extension method

SeanFarrow opened the pull request:
This PR solves issue #539.

moodmosaic commented on Src/AutoFixtureUnitTest/FixtureTest.cs:
```diff
@@ -1847,63 +1847,7 @@ public void DefaultRepeatCountIsThree()
             Assert.Equal<int>(expectedRepeatCount, result);
             // Teardown
         }
-
```
Wouldn't it be possible to avoid changing/moving these tests?

AFAICT, this change is a pure refactoring, it shouldn't be necessary to modify existing tests. This also makes it possible to verify that no breaking changes are introduced.

Since `Repeat` is extracted into an `IFixture` extension method, `sut` should still be able to access it, so these tests should still compile and pass.

What do I probably missed? :flushed:

moodmosaic commented on Src/AutoFixture/IFixtureExtensions.cs:
```diff
@@ -0,0 +1,40 @@
+
+namespace Ploeh.AutoFixture
+{
+    public static class IFixtureExtensions
```
Inspired by Haskell's [replicate](http://zvon.org/other/haskell/Outputprelude/replicate_f.html), how about naming this `FixtureReplicator`, `Replicator`, or similar? Or maybe we could name the class `FixtureRepeater`, `Repeater`, and so on.

FWIW, we have another class with extension methods, and it's called [`FixtureFreezer`](https://github.com/AutoFixture/AutoFixture/blob/master/Src/AutoFixture/FixtureFreezer.cs) instead of the general purpose XyzExtensions name.

moodmosaic replied:
I think we need some XML documentation there, otherwise the Code Analysis treats this as a warning (which then gets treated as an error during the build process).

moodmosaic commented:
Thank you for opening a Pull Request!  :smiley: :+1:

I just had a look and added a few comments there.

SeanFarrow commented:
Agree with all your comments, will fix later.

 

From: Nikos Baxevanis [mailto:notifications@github.com] 
Sent: 07 March 2016 07:22
To: AutoFixture/AutoFixture <AutoFixture@noreply.github.com>
Cc: Sean Farrow <s.r.farrow@icloud.com>
Subject: Re: [AutoFixture] Move Repeat<t> to an extension method (#540)

 

Thank you for opening a Pull Request!   <https://assets-cdn.github.com/images/icons/emoji/unicode/1f603.png>   <https://assets-cdn.github.com/images/icons/emoji/unicode/1f44d.png> 

I just had a look and added a few comments there.

—
Reply to this email directly or view it on GitHub <https://github.com/AutoFixture/AutoFixture/pull/540#issuecomment-193134780> .  <https://github.com/notifications/beacon/ABY1fh37fwnVCvb54xOp8hFAo-E5dlP5ks5pq9IngaJpZM4HqRdf.gif>

SeanFarrow commented:
Ok, All comments have been addressed.

I’ve moved this to a new branch FixtureRepeater to more closely align with what the pr does.

 

 

From: Nikos Baxevanis [mailto:notifications@github.com] 
Sent: 07 March 2016 07:22
To: AutoFixture/AutoFixture <AutoFixture@noreply.github.com>
Cc: Sean Farrow <s.r.farrow@icloud.com>
Subject: Re: [AutoFixture] Move Repeat<t> to an extension method (#540)

 

Thank you for opening a Pull Request!   <https://assets-cdn.github.com/images/icons/emoji/unicode/1f603.png>   <https://assets-cdn.github.com/images/icons/emoji/unicode/1f44d.png> 

I just had a look and added a few comments there.

—
Reply to this email directly or view it on GitHub <https://github.com/AutoFixture/AutoFixture/pull/540#issuecomment-193134780> .  <https://github.com/notifications/beacon/ABY1fh37fwnVCvb54xOp8hFAo-E5dlP5ks5pq9IngaJpZM4HqRdf.gif>

ploeh commented:
@SeanFarrow, you closed this pull request by yourself, but from the comments I get the impression that you've addressed the review somewhere else. If you'd like us to review and ultimately pull those changes, you'll have to open a new pull request.

If you do that, I'll be happy to consider it.

SeanFarrow commented:
I’ve opened a new pr!

 

 

From: Mark Seemann [mailto:notifications@github.com] 
Sent: 12 March 2016 16:37
To: AutoFixture/AutoFixture <AutoFixture@noreply.github.com>
Cc: Sean Farrow <s.r.farrow@icloud.com>
Subject: Re: [AutoFixture] Move Repeat<t> to an extension method (#540)

 

@SeanFarrow <https://github.com/SeanFarrow> , you closed this pull request by yourself, but from the comments I get the impression that you've addressed the review somewhere else. If you'd like us to review and ultimately pull those changes, you'll have to open a new pull request.

If you do that, I'll be happy to consider it.

—
Reply to this email directly or view it on GitHub <https://github.com/AutoFixture/AutoFixture/pull/540#issuecomment-195770080> .  <https://github.com/notifications/beacon/ABY1foSSzxOmFb7DRHZ3fAMs7X_CF9dfks5psuuPgaJpZM4HqRdf.gif>
